utils/config.go

```diff
@@ -434,6 +436,17 @@ func (config *Config) LoadConfigFile(configFile string) error {
 	if err != nil {
 		return fmt.Errorf("Error at JSON parsing config document: %v", err)
 	}
+
+	if len(config.ServiceConfig.TempDir) > 0 {
+		if _, err := os.Stat(config.ServiceConfig.TempDir); os.IsNotExist(err) {
+			log.Printf("Creating temp directory: %v", config.ServiceConfig.TempDir)
+			err := os.MkdirAll(config.ServiceConfig.TempDir, os.ModePerm)
```

Comment:
`os.MkdirAll` checks for directory existence, making the `os.Stat` call redundant. See https://golang.org/pkg/os/#MkdirAll